src/main/java/games/dicemonastery/DiceMonasteryTurnOrder.java

```diff
@@ -0,0 +1,321 @@
+package games.dicemonastery;
+
+import core.AbstractGameState;
+import core.CoreConstants;
+import core.actions.AbstractAction;
+import core.components.Component;
+import core.turnorders.TurnOrder;
+import games.dicemonastery.actions.LogEvent;
+
+import java.util.*;
+
+import static games.dicemonastery.DiceMonasteryConstants.*;
+import static games.dicemonastery.DiceMonasteryConstants.ActionArea.*;
+import static games.dicemonastery.DiceMonasteryConstants.Phase.BID;
+import static games.dicemonastery.DiceMonasteryConstants.Phase.SACRIFICE;
+import static games.dicemonastery.DiceMonasteryConstants.Season.*;
+import static java.util.stream.Collectors.*;
+
+public class DiceMonasteryTurnOrder extends TurnOrder {
+
+    public DiceMonasteryTurnOrder(int nPlayers, DiceMonasteryParams params) {
+        super(nPlayers, params.YEARS);
+    }
+
+    private DiceMonasteryTurnOrder(int nPlayers) {
+        super(nPlayers);
+    }
+
+    Season season = SPRING;
```

Comment:
This seems like a game state variable, not turn order?

Reply:
Interesting point. I have treated TurnOrder as encapsulating all of the state associated with what turn it is (so the GameState does not store this, but just uses the internal TurnOrder object for this), as well as the logic of when to move the turn forard / change the current player.
I can move this out if it breaks a TAG design principle.